Add JS release notes for July 2020 (#1506)
Azure/azure-sdk: releases/2020-07/js.md

#### GA

> - *TODO*

Author: Do we have any lib graduating to GA this time, search? @sarangan12

Reviewer: Yes, Search is going GA

Author: Good! I moved Search to the top for visibility.

#### GA

- Azure Cognitive Search

#### Updates

- Azure App Configuration
- Core libraries
- Azure Event Hubs

#### Preview

- Azure Event Hubs
- Azure Cognitive Form Recognizer
- Azure Key Vault Certificates/Keys/Secrets
- Azure Service Bus

## Installation Instructions

To install the packages, copy and paste the below into a terminal.

```bash
$> npm install @azure/app-configuration
$> npm install @azure/event-hubs

> $> npm install @azure/event-hubs@next

Reviewer: Above we say "copy and paste the below into a terminal`. Why would one install event hubs latest followed by next right after?

Author: Yes I am aware of this. I need to adjust to account for Event Hubs' update and preview at the same time.

Detailed changelogs are linked from the [Quick Links](#quick-links) below. Here are some of the highlights:

### App Configuration

#### New Features

> - *TODO* add browser support

Reviewer 2: ```suggestion
- Adding browser support for the latest versions of Chrome, Edge and Firefox.
```

### Event Hubs

> Updates to last stable release

Author: Any better way to list both GA update and preview for Event Hubs?

#### New Features

- *TODO*

#### Breaking Changes

- *TODO*

#### Key Bug Fixes

- *TODO*

New preview version

#### New Features

- *TODO*

#### Breaking Changes from Last Preview

- *TODO*

#### Key Bug Fixes

- *TODO*

### Identity

#### New Features

- *TODO*

#### Breaking Changes from Last Preview

- *TODO*

#### Key Bug Fixes

- *TODO*

### Key Vault Certificates/Keys/Secrets

#### New Features

- *TODO*

#### Breaking Changes from Last Preview

- *TODO*

#### Key Bug Fixes

- *TODO*

### Form Recognizer

#### New Features

- *TODO*

#### Breaking Changes from Last Preview

- *TODO*

#### Key Bug Fixes

- *TODO*

### Search

#### New Features

- *TODO*

#### Breaking Changes from Last Preview

- *TODO*

#### Key Bug Fixes

- *TODO*

### Service Bus

#### New Features

> - *TODO*

Reviewer 2: ```suggestion
- (Receiver|SessionReceiver).subscribe() now returns a closeable object which will stop new messages from arriving but still leave the receiver open so they can be settled via methods like complete().
```

Reviewer 2: @HarshaNalluru - I believe we do a summary here rather than a point by point but I leave it up to you to summarize your life's work.

Reviewer 3: ```suggestion
- Adds abortSignal support throughout Sender and non-session Receivers.
- (Receiver|SessionReceiver).subscribe() now returns a closeable object which will stop new messages from arriving but still leave the receiver open so they can be settled via methods like complete().
- `OperationOptions` support has been added for the methods under `ServiceBusManagementClient`, this adds support for abortSignal, requestOptions when creating and sending HTTP requests.
```
(Includes @richardpark-msft's suggestion too.)

Author: `a closeable object which will stop new messages from arriving`
will it stop new messages from arriving, or `can be used to stop `?

Reviewer 2: I think it's accurate as it is but it's perhaps not as straightforward as I'd like - it does stop new messages from arriving _but_ your handlers do continue to run (and any messages that arrive between the actual "stopping" will still get delivered to you.

Author: hmm, I am surprised that the name `subscribe()` would be related to "stop". I will merge and you can add new suggestion please?

#### Breaking Changes from Last Preview

- *TODO*

#### Key Bug Fixes

> - *TODO*

Reviewer 2: @HarshaNalluru, I believe this is definitely your section. :)

Reviewer 3: ```suggestion
- Fixed the bug where the messages scheduled in parallel with the `scheduleMessage` method have the same sequence number in response.
- Fixed the bug where the `userProperties` in a correlation filter are not populated in the rule while using the `createRule()` method.
```

> - *TODO*

Reviewer 3: ```suggestion
- Standardized methods on senders and receivers to use the `Messages` suffix and deal with multiple messages rather than have dedicated methods to deal with a single message. 
- Standardized methods that peek and receive a given number of messages to use a similar signature. 
- Removed `isReceivingMessages` method on the `Receiver`.
- Management api updates

  - Renamed `createdOn`, `accessedOn` and `modifiedOn` properties to `createdAt`, `accessedAt` and `modifiedAt`, updated the corresponding type from `ISO-8601 timestamp string` to the `Date` type in the responses for the `runtimeInfo` methods for Queue, Topic and Subscription.
  - The property `top` in the options passed to any of the methods that get information for multiple entities like `getQueues` or `getQueuesRuntimeInfo` is renamed to `maxCount`.
  - The "update" methods (`updateQueue`, `updateTopic` and `updateSubscription`) now require all properties on the given queue/topic/subscription object to be set though only a subset of them are actually updatable. Therefore, the suggested flow is to use the "get" methods to get the queue/topic/subscription object, update as needed, and then pass it to the "update" methods.

```

Author: Are the first two basically
- Removed methods that deal with single messages.
- Added `Messages` suffix to methods dealing with multiple messages. For example, ...

Reviewer 3: > Are the first two basically
Removed methods that deal with single message.
Added Messages suffix to methods dealing with multiple messages. For example, ...

That's just the first point.

```suggestion
- Standardized methods on senders and receivers to use the `Messages` suffix, removed dedicated methods dealing with a single message to reduce API surface.
- Standardized methods that peek and receive a given number of messages to use a similar signature. 
  Old: `peekMessages(options);` and `receiveMessages(maxMessages, options)`
  New: `peekMessages(maxMessageCount, options);` and `receiveMessages(maxMessageCount, options)`
- Removed `isReceivingMessages` method on the `Receiver`.
- Management api updates

  - Renamed `createdOn`, `accessedOn` and `modifiedOn` properties to `createdAt`, `accessedAt` and `modifiedAt`, updated the corresponding type from `ISO-8601 timestamp string` to the `Date` type in the responses for the `runtimeInfo` methods for Queue, Topic and Subscription.
  - The property `top` in the options passed to any of the methods that get information for multiple entities like `getQueues` or `getQueuesRuntimeInfo` is renamed to `maxCount`.
  - The "update" methods (`updateQueue`, `updateTopic` and `updateSubscription`) now require all properties on the given queue/topic/subscription object to be set though only a subset of them are actually updatable. Therefore, the suggested flow is to use the "get" methods to get the queue/topic/subscription object, update as needed, and then pass it to the "update" methods.
```

Re-worded the first point and added the changes to the second point, does this help @jeremymeng ? or Am I just adding to the confusion?

Author: Yes it helped. I'd thought the second point were similar to first point

Detailed changelogs are linked from the [Quick Links](#quick-links) below. Here are some of the highlights:

### Search

This is the first stable release version of the Cognitive Search library.

#### New Features

- *TODO*

#### Breaking Changes from Last Preview

- *TODO*

#### Key Bug Fixes

- *TODO*

### App Configuration

#### New Features

- Added browser support for the latest versions of Chrome, Edge and Firefox.

### Event Hubs

Updates to last stable release

> #### New Features
> 
> - *TODO*
> 
> #### Breaking Changes
> 
> - *TODO*
> 
> #### Key Bug Fixes
> 
> - *TODO*

Reviewer 4: ```suggestion
#### Key Bug Fixes

- Improved the reliability of closing `EventHubConsumerClient` and `Subscriptions`.
```

New preview version

> #### New Features
> 
> - *TODO*
> 
> #### Breaking Changes from Last Preview
> 
> - *TODO*
> 
> #### Key Bug Fixes
> 
> - *TODO*
> 

Reviewer 4: ```suggestion
#### New Features

- The `EventHubConsumerClient` now supports configuring `loadBalancingOptions` for more control over performance tuning while load balancing.
```

Detailed changelogs are linked from the [Quick Links](#quick-links) below. Here are some of the highlights:

### Search

This is the first stable GA release of the Cognitive Search library.

#### Breaking Changes from Last Preview

- In Suggest API & Search API return values, a new property called `document` is introduced. All user-defined fields are moved inside this `document` property.
- In `analyzeText` API, the `text` parameter is moved from method level to inside `options` bag.
- In `search` API, `includeTotalResultCount` property is renamed to `includeTotalCount`.
- In `ServiceCounters`, the `skillsetCounter` property has been removed.
- Modified the names of several properties. Please refer [#9321](https://github.com/Azure/azure-sdk-for-js/issues/9321) for a detailed list of renames.

### App Configuration

#### New Features

- Added browser support for the latest versions of Chrome, Edge and Firefox.

### Event Hubs

> We have released two versions of the Event Hubs library this time: updates to the stable version, and a new preview version.
> 
> Updates to last stable release
> 
> #### Key Bug Fixes
> 
> - Improved the reliability of closing `EventHubConsumerClient` and `Subscriptions`.
> 
> New preview version
> 
> #### New Features
> 
> - The `EventHubConsumerClient` now supports configuring `loadBalancingOptions` for more control over performance tuning while load balancing.

Reviewer: ```suggestion
We have released two versions of the Event Hubs library this time. An update to the stable version with bug fixes, and and a preview version with new features.

#### Key Bug Fixes

- The update to the stable version i.e version 5.2.2 has improved the application reliability when closing `EventHubConsumerClient` and `Subscriptions`.

#### New Features

- The preview version 5.3.0-preview.1 now supports configuring `loadBalancingOptions` when constructing the `EventHubConsumerClient` for more control over performance tuning while load balancing.
```

Author: Looks better!  There is an extra `and`.  I will remove it.
